internal/cni/cni.go

```diff
@@ -347,7 +348,24 @@ func CmdDel(args *skel.CmdArgs) error {
 
 	if !cfg.SkipUnloadBpf {
 		logging.Infof("cmdDel(): removing BPF program from device")
-		if err := bpfHandler.Cleanbpf(cfg.Device); err != nil {
+
+		device, err := netlink.LinkByName(cfg.Device)
+		if err != nil {
+			err = fmt.Errorf("cmdDel(): failed to find device %q in containerNS: %w", cfg.Device, err)
+			logging.Errorf(err.Error())
+
+			return err
+		}
+		logging.Infof("cmdDel(): setting link to down state and attached to false")
+		device.Attrs().Xdp.Attached = false
+		if err := netlink.LinkSetDown(device); err != nil {
+			err = fmt.Errorf("cmdDel(): error setting device to down state: %w", err)
+			logging.Errorf(err.Error())
+
+			return err
+		}
+		logging.Infof("cmdDel():link is set to down state and attached is false")
+		if err := netlink.LinkSetXdpFd(device, -1); err != nil {
```

Comment:
I think the changes here in cni.go might be coming from the time we tried to cleanup using netlink to remove the bpf app?

I'd imagine a move to libxdp should not require changes to the cni code, just our bpf code.